packages/prime-field-group/src/ui/InputComponent.tsx

```diff
@@ -46,7 +48,8 @@ export class InputComponent extends React.PureComponent<PrimeFieldProps, any> {
   }
 
   public componentWillReceiveProps(nextProps: PrimeFieldProps) {
-    if (!this.props.document && nextProps.document) {
+    if ((!this.props.document && nextProps.document) ||
+      JSON.stringify(this.props.initialValue) !== JSON.stringify(nextProps.initialValue)) {
```

Comment:
? need to json it? why? 
in react the props should be immutable, so should not be necessary

Reply:
yeah, can remove the JSON.stringify, but still need the check `this.props.initialValue !== nextProps.initialValue`

Reply:
yeah thats fine, this additional initialValue check is because of the update?